src/nvim/tag.c

```diff
@@ -1254,15 +1252,7 @@ static int find_tagfunc_tags(
       break;
     }
 
-    if (name_only) {
-      mfp = vim_strsave(res_name);
-    } else {
-      mfp = (char_u *)xmalloc((int)sizeof(char_u) + len + 1);
```

Comment:
When is `sizeof(char_u)` not equal to 1 in C99?

Reply:
regardless, sizeof() in malloc() is just good practice to show the intent (though in Nvim codebase we would use `sizeof(*foo)`